pkg/scene/widget.go

```diff
@@ -54,6 +72,14 @@ func (w *Widget) Enabled() bool {
 	return w.enabled
 }
 
+func (w *Widget) WidgetLayout() *WidgetLayout {
+	return w.widgetLayout
+}
+
 func (w *Widget) SetEnabled(enabled bool) {
 	w.enabled = enabled
 }
+
+func (w *Widget) SetExtended(extended *bool) {
+	w.widgetLayout.Extended = extended
```

Comment:
extended should be copied to prevent unexpected changes from outside

Reply:
Sorry what do you mean exactly by it should be copied? Do you mean the param shouldn't be a pointer? Or do you mean I should set it to a local variable and assign the local variable to w.widgetLayout.Extended? Or something else?

Reply:
```go
e := true
e2 := &e
w.SetExtended(e2)
*e2 = false // This means that you can manipulate the field without going through the
```

Reply:
Ah, so you are saying I don't need this method? And I should just directly manipulate extended by copying?

Reply:
I mean `extended` needs to be copied in the method:

```go
func (w *Widget) SetExtended(extended *bool) {
	if extended == nil {
		w.widgetLayout.Extended = nil
	} else {
		extended2 := *extended
		w.widgetLayout.Extended = &extended2
	}
```

Reply:
Sorry, thank you! Time to read up on dereferencing and when/why it is good!